### ISSUE TITLE
wrong-type-argument package-desc error when using extended dependency format

### ISSUE THREAD

**Issue by suhail-singh** [emacs-eldev/eldev#106]
**Describe the bug**

**Steps to reproduce**

1. Take a project that uses `eldev` and has `melpa` archive enabled.
2. Add `(eldev-add-extra-dependencies 'build '(:package magit))` to `Eldev` file.
3. Run `eldev -dt dependencies build`.
4. See error.

**Expected behavior**

No exception should be raised.  Per the documentation of "Extended dependency
format", `eldev-add-extra-dependencies` should accept a plist.

**Actual behavior**

``` shell
...
Debugger entered--Lisp error: (wrong-type-argument package-desc magit)
  signal(wrong-type-argument (package-desc magit))
  eldev-message-version(magit t)
  eldev-dependencies("build")
  apply(eldev-dependencies "build")
  eldev--execute-command(("dependencies" "build"))
  eldev-cli(("-dt" "dependencies" "build"))
...
```

**Environment**

- Eldev: 1.10.3
- Emacs: 29.4
- OS: Linux

**Additional context**

N/A.

**Comment by doublep** [comment]
A problem is that `eldev-add-extra-dependencies` predates the extended dependency format. Moreover, it accepts lists as dependencies, but for a different meaning. I guess I need to fix this anyway, but it would be useful to understand what exactly you are trying to achieve. Is this only a neatpick (as `(:package magit)` should be the same as just `magit`), or were you planning to use some functionality supported only by the extended format?

**Comment by doublep** [comment]
Fix will be released with 1.11. Apparently, the bug was only with commands `dependencies` and `dependency-tree`, otherwise the extended format worked.

**Comment by doublep** [comment]
Fixed in 1.11.
